devops/roles/postgresql-restore/tasks/main.yml

```diff
@@ -0,0 +1,40 @@
+- name: ensure restore dir exists
+  file: path="{{ postgresql_restore_dir }}" state=directory owner={{ postgresql_user }} group={{ postgresql_user }}
+
+- set_fact:
+    postgresql_restore_gzip_file_path: "{{ postgresql_restore_dir }}/{{ postgresql_restore_gzip_file_name }}"
+
+- name: Download restore file from azure
+  command: az storage blob download --container-name {{ postgresql_restore_azure_container_name }} --name {{ postgresql_restore_gzip_file_name }} --file {{ postgresql_restore_gzip_file_path }}
+  environment:
+    AZURE_STORAGE_ACCOUNT: "{{ postgresql_restore_azure_storage_account_name }}"
+    AZURE_STORAGE_KEY: "{{ postgresql_restore_azure_storage_access_key }}"
+  async: 3600
+  poll: 10
+
+- name: ensure postgresql service is stopped
+  service: name=postgresql state=stopped
+
+- name: wait for postgresql to be stopped
+  wait_for: port={{ postgresql_port }} state=stopped
+
+- name: drop cluster
+  command: pg_dropcluster {{ postgresql_cluster_version }} {{ postgresql_cluster_name }}
```

Comment:
Does restoring the postgres backup always requires the cluster to be recreated?

Reply:
This will be required when we restore the backup in a new machine. Resolving this comment.